### PR TITLE
Display error if user tries to rename file with a slash

### LINKED ISSUE #946: file can be moved by renaming it with a `/` in its name
### Steps to reproduce
1. have a file "file1" and a folder "folder"
2. rename "file1" to `folder/file1`

### Expected behaviour
error shown that the name is invalid

### Actual behaviour
file is moved into the folder

we had this also in core see https://github.com/owncloud/core/issues/28475
and it was decided that its a bug not a feature :-(

### PR DESCRIPTION
**Pull request by LukasHirt**
## Description
Added condition to prevent renaming with a slash.

## Related Issue

- Fixes #946

## How Has This Been Tested?

- test environment: manually

## Screenshots (if appropriate):
image

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

File: apps/files/src/mixins.js [hunk 1]
```diff
@@ -45,11 +48,18 @@ export default {
...
+        if (!item.includes('/')) {
+          this.renameFile({
+            client: this.$client,
+            file: this.selectedFile,
+            newValue: item
+          })
+        } else {
+          this.showNotification({
+            title: this.$gettext(`Rename of ${this.selectedFile.name} failed - can not contain "/"`),
+            type: 'error'
```
Comment (DeepDiver1975) [inline apps/files/src/mixins.js]: Unrelated to this change: type error should display the notification in red .... something is broken ....

File: apps/files/src/mixins.js [hunk 2]
```diff
@@ -45,11 +48,18 @@ export default {
         this.selectedFile = item
         this.newName = item.name
       } else {
-        this.renameFile({
-          client: this.$client,
-          file: this.selectedFile,
-          newValue: item
-        })
+        if (!item.includes('/')) {
```
Comment (DeepDiver1975) [inline apps/files/src/mixins.js]: General best practice: use positive conditions
```
if (item.includes('/')) {
  // show error
}
....

File: apps/files/src/mixins.js [hunk 3]
```diff
@@ -45,11 +48,18 @@ export default {
...
+          this.renameFile({
+            client: this.$client,
+            file: this.selectedFile,
+            newValue: item
+          })
+        } else {
```
Comment (DeepDiver1975) [inline apps/files/src/mixins.js]: General best practice: simplify control structure by returning early .... e.g.

```
if (item.includes()) {
  showNotification(...)
  return
}

this.rename(....)

File: apps/files/src/mixins.js [hunk 4]
```diff
@@ -45,11 +48,18 @@ export default {
...
+          this.showNotification({
+            title: this.$gettext(`Rename of ${this.selectedFile.name} failed - can not contain "/"`),
```
Comment (DeepDiver1975) [inline apps/files/src/mixins.js]: Should be 'Renaming of ... failed. The file name cannot contain a "/"

@settermjd for native :tongue: :wink:

File: apps/files/src/mixins.js (hunk 1 above)
Comment (LukasHirt) [reply apps/files/src/mixins.js]: I looked at it and the problem is in the name of type prop - should be status. I guess it's still after vuetify maybe and after creating ods component it wasn't changed.

File: apps/files/src/mixins.js (hunk 2 above)
Comment (LukasHirt) [reply apps/files/src/mixins.js]: Fixed

File: apps/files/src/mixins.js (hunk 3 above)
Comment (LukasHirt) [reply apps/files/src/mixins.js]: Fixed

File: apps/files/src/mixins.js (hunk 4 above)
Comment (LukasHirt) [reply apps/files/src/mixins.js]: Fixed

File: apps/files/src/mixins.js [hunk 5]
```diff
@@ -45,11 +48,19 @@ export default {
...
-          file: this.selectedFile,
-          newValue: item
-        })
+        if (item.includes('/')) {
+          this.showNotification({
+            title: this.$gettext(`Renaming of ${this.selectedFile.name} failed. The file name cannot contain a "/"`),
+            status: 'danger'
+          })
+          return
+        } else {
```
Comment (DeepDiver1975) [inline apps/files/src/mixins.js]: remove the else - THX

Comment (LukasHirt) [reply apps/files/src/mixins.js]: Ah, right - fixed.

**Pull request merged**